spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceWebFilter.java

```diff
@@ -111,6 +116,14 @@ Tracer tracer() {
 		return this.tracer;
 	}
 
+	CurrentTraceContext currentTraceContext() {
+		if (this.currentTraceContext == null) {
+			this.currentTraceContext = this.beanFactory.getBean(HttpTracing.class)
```

Comment:
I think we only need the currentTraceContext, which has less strange wiring loop potential. Sometimes people use brave internally to make the Httptracing object. In some extreme cases, we guard the whole thing https://github.com/spring-cloud/spring-cloud-sleuth/blob/bd149ce4a7c2154fe8526394d14b6ab89addd62e/spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/ReactorSleuth.java#L76-L79

Reply:
sorry, do not get it.  
what do you propose?